mkosi/__main__.py

```diff
@@ -24,8 +24,8 @@ def main() -> None:
                     run_verb(a)
             else:
                 run_verb(a)
-    except MkosiException:
-        sys.exit(1)
+    except MkosiException as e:
+        sys.exit(e.returncode)
```

Comment:
Not sure how valuable that information is, since we can't really expect from where this comes and this could be any number. Maybe this could print the failing cmdline (or binary as well).

Reply:
The behavior I'm going for here is that when running "shell", "boot" or "qemu", mkosi behaves as an invisible wrapper around the underlying command we're actually running (systemd-nspawn or qemu). I don't think we should show any extra output by default either (unless we have to build the image of course).

Anyway, for "shell", propagating the return code seems useful, since you can then use the "shell" verb in scripting to run commands and check the returncode for additional failure information from the command you tried to run since nspawn propagates that information in that scenario.

Reply:
Ah, I see, from that point of view this makes a lot of sense. How about putting the additional metainformation in the exception and only showing it when we actually do a build (or rather, as a first iteration, here only checking if the command is `build`)?

Reply:
I'm not sure I understand what you're trying to say here. I'd start with just propagating exit codes since silencing the output will take a bit more thinking since there's quite a few useful warnings in the qemu path. Maybe we should just make sure we don't print any output on the "shell" path (which we already do) and leave the qemu output for what it is.

Reply:
Sorry for being unclear. What I meant was adding adding a `cmdline` or `args` attribute to `MkosiException` as well and in main, additionally to propagating the returncode, also printing the cmdline if we are not doing a build. Since that might be too cumbersome to decide right now (since some commands imply build), I was thinking something along the lines
```python
except MkosiException as e:
    if args.verb == "build"
        print(e.cmdline)
    sys.exit(e.returncode)
```

That being said, I just remembered, that maybe we don't need to decorate `MkosiException` with more attributes, that are already in the original exception. I'll add another comment at the appropriate point in the code.

Reply:
Ah, now I get it. I had forgotten this wasn't in the build path (just looking on this on the go, so looking at the greater picture is a bit difficult). Nevertheless, grabbing the information from the initial exception as described below might still be worthwhile, your call. :)